newIDE/app/src/InstancesEditor/index.js

```diff
@@ -113,9 +113,13 @@ export default class InstancesEditorContainer extends Component {
       this.props.height
     );
     gesture.panable(this.backgroundArea);
-    this.backgroundArea.on('mousedown', event =>
-      this._onBackgroundClicked(event.data.global.x, event.data.global.y)
-    );
+    this.backgroundArea.on('mousedown', event => {
+      if (event.data.button === 1 || this.keyboardShortcuts.shouldMoveView()) {
```

Comment:
Ideally we want to avoid messing with event propagation and hardcoding buttons there: most of the event handling functions are just redirecting to other functions (like _onBackgroundClicked) that are "higher level" and are supposed to understand what's going on. 

In this case, when the user does the action of moving the mouse (or the touch) and drag it, it's called a "pan". At the end, _onPanEnd is called, meaning that this action of click + move is finished. Two possibilities: either the user moved the view (`shouldMoveView()` returns true) or a selection rectangle was finished.

If you look at `_onPanEnd`, you can see that there is a comment about this, and a check:
`if (this.selectionRectangle.hasStartedSelectionRectangle())`

Unfortunately, it seems that this check if true, leading to a call to `selectInstances`, which select... 0 instances as the selection rectangle contains nothing.

So now the question is, why `hasStartedSelectionRectangle` returned true?
Turns out that as soon as you click on the background, `_onBackgroundClicked` is called. (which is normal) 

Inside this function, you can see that a selection rectangle is always started: 

https://github.com/4ian/GDevelop/blob/657ca853240eae806f3d04353a20e693d64fe0d9/newIDE/app/src/InstancesEditor/index.js#L413

This might be the culprit! Add a `if (!this.keyboardShortcuts.shouldMoveView()) {  }` around it.
Now the click on the background should properly start a selection ONLY if the view is not being moved. So it won't be messing with the pan ending (`_onPanEnd`) which was wrongly thinking a selection rectangle was drawn :)

This seems actually be a regression introduced recently when I moved the selection rectangle start into `_onBackgroundClicked` to ensure the starting point is always correct even in case of lag.